src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/IdentityRepository.php

```diff
@@ -100,6 +100,26 @@ public function findByNameIdsIndexed(array $nameIds)
     /**
      * @param NameId      $nameId
      * @param Institution $institution
+     *
+     * @return bool
+     */
+    public function hasIdentityWithNameIdAndInstitution(NameId $nameId, Institution $institution)
+    {
+        $identityCount = $this->createQueryBuilder('i')
+            ->select('COUNT(i.id)')
```

Comment:
Not that important, but I'm not sure if a COUNT would be the optimal solution here. Perhaps a `SELECT *` with a `LIMIT 1` clause would be faster.